### PR TITLE
enable external resource gc

### PR DESCRIPTION
**Pull request by bdehri**
### Checklist

- [x] Update changelog in CHANGELOG.md.

### Trigger e2e tests

/test create
/test upgrade

### CODE REVIEW

**Review by fiunchinho — COMMENTED** [review]
Is there a reason not to enable this? I'd always enable it and don't add it to the helm values. What do you think?

File: helm/cluster-aws/values.yaml [hunk 1]
```diff
@@ -8,6 +8,7 @@ baseDomain: "example.com"
 aws:
   region: ""
   awsClusterRole: "default"
+  externalResouceGc: true
```
Comment (calvix) [inline helm/cluster-aws/values.yaml:11]: typo

Comment (calvix) [reply helm/cluster-aws/values.yaml:11]: can we add commetn above the value to explain what it does?

File: helm/cluster-aws/templates/_aws_cluster.tpl [hunk 2]
```diff
@@ -11,6 +11,7 @@ metadata:
     {{- if (eq .Values.network.vpcMode "private") }}
     cluster.x-k8s.io/paused: "true"
     {{end}}
+    aws.cluster.x-k8s.io/external-resource-gc: "{{ .Values.aws.externalResourceGc }}"
```
Comment (calvix) [inline helm/cluster-aws/templates/_aws_cluster.tpl:14]: where is this annotation came from?

**Comment by bdehri** [comment]
> Is there a reason not to enable this? I'd always enable it and don't add it to the helm values. What do you think?

No specific reason actually, it is a little bit verbose in this way IMHO

**Review by calvix — APPROVED** [review]

**Comment by calvix** [comment]
> Is there a reason not to enable this? I'd always enable it and don't add it to the helm values. What do you think?

yeah true, we always want to clean external resources, so maybe add it to the annotations, and no need to add it as a configurable value :p at least we can avoid a situation where someone disables it without thinking

**Comment by bdehri** [comment]
@fiunchinho I saw you comment about these failing tests are ignorable. Is this still the case?

**Comment by fiunchinho** [comment]
> @fiunchinho I saw your comment about these failing tests are ignorable. Is this still the case?

There were broken yesterday but I just fixed them. Let's try!

/test create
/test upgrade

**Pull request merged**